File: frontend/src/types/character.ts
/**
 * 人物设定系统 - TypeScript 类型定义
 *
 * 提供人物角色相关的类型定义
 */

// ==================== 枚举定义 ====================

export type CharacterLevel = 'protagonist' | 'major_support' | 'support' | 'minor' | 'past';
export type CharacterGender = 'male' | 'female' | 'other' | 'unknown';
export type AliasType = 'zi' | 'hao' | 'nickname' | 'title' | 'other';
export type RelationType = 'family' | 'love' | 'friend' | 'master' | 'apprentice' | 'enemy' | 'other';
export type ArtifactType = 'weapon' | 'armor' | 'accessory' | 'treasure' | 'other';
export type ArtifactRarity = 'legendary' | 'epic' | 'rare' | 'common';

// 角色等级显示名称
export const CharacterLevelLabels: Record<CharacterLevel, string> = {
  protagonist: '主角',
  major_support: '重要配角',
  support: '配角',
  minor: '小角色',
  past: '过往',
};

// 角色等级颜色配置
export const CharacterLevelColors: Record<CharacterLevel, { border: string; bg: string; bar: string; barWidth: number }> = {
  protagonist: {
    border: 'hsl(0 60% 50%)',
    bg: 'hsl(0 60% 97%)',
    bar: 'hsl(0 60% 50%)',
    barWidth: 4,
  },
  major_support: {
    border: 'hsl(45 50% 50%)',
    bg: 'hsl(45 50% 97%)',
    bar: 'hsl(45 50% 50%)',
    barWidth: 3,
  },
  support: {
    border: 'hsl(210 30% 55%)',
    bg: 'hsl(210 30% 97%)',
    bar: 'hsl(210 30% 55%)',
    barWidth: 2,
  },
  minor: {
    border: 'hsl(0 0% 60%)',
    bg: 'hsl(0 0% 98%)',
    bar: 'hsl(0 0% 60%)',
    barWidth: 2,
  },
  past: {
    border: 'hsl(280 30% 55%)',
    bg: 'hsl(280 30% 97%)',
    bar: 'hsl(280 30% 55%)',
    barWidth: 2,
  },
};

// 角色等级尺寸配置
export const CharacterLevelSizes: Record<CharacterLevel, { height: number; padding: number; avatar: number; radius: number }> = {
  protagonist: { height: 100, padding: 14, avatar: 68, radius: 8 },
  major_support: { height: 90, padding: 12, avatar: 62, radius: 7 },
  support: { height: 82, padding: 12, avatar: 54, radius: 6 },
  minor: { height: 74, padding: 10, avatar: 48, radius: 5 },
  past: { height: 74, padding: 10, avatar: 48, radius: 5 },
};

// 性别显示名称
export const CharacterGenderLabels: Record<CharacterGender, string> = {
  male: '男',
  female: '女',
  other: '其他',
  unknown: '未知',
};

// 别名类型显示名称
export const AliasTypeLabels: Record<AliasType, string> = {
  zi: '字',
  hao: '号',
  nickname: '外号',
  title: '称号',
  other: '其他',
};

// 关系类型显示名称
export const RelationTypeLabels: Record<RelationType, string> = {
  family: '亲情',
  love: '爱情',
  friend: '友情',
  master: '师父',
  apprentice: '徒弟',
  enemy: '敌对',
  other: '其他',
};

// 关系类型颜色
export const RelationTypeColors: Record<RelationType, string> = {
  family: 'hsl(120 40% 50%)',
  love: 'hsl(330 70% 60%)',
  friend: 'hsl(200 60% 55%)',
  master: 'hsl(270 50% 55%)',
  apprentice: 'hsl(280 45% 50%)',
  enemy: 'hsl(0 60% 50%)',
  other: 'hsl(0 0% 60%)',
};

// 器物类型显示名称
export const ArtifactTypeLabels: Record<ArtifactType, string> = {
  weapon: '武器',
  armor: '防具',
  accessory: '饰品',
  treasure: '法宝',
  other: '其他',
};

// 器物等级显示名称
export const ArtifactRarityLabels: Record<ArtifactRarity, string> = {
  legendary: '神器',
  epic: '传说',
  rare: '稀有',
  common: '普通',
};

// 器物等级边框颜色（低饱和度）
export const ArtifactRarityColors: Record<ArtifactRarity, string> = {
  legendary: 'hsl(0 60% 50%)',   // 红色 - 神器
  epic: 'hsl(45 50% 50%)',       // 金色 - 传说
  rare: 'hsl(210 30% 55%)',      // 银色 - 稀有
  common: 'hsl(0 0% 60%)',       // 灰色 - 普通
};

// 器物等级背景颜色
export const ArtifactRarityBgColors: Record<ArtifactRarity, string> = {
  legendary: 'hsl(0 60% 97%)',   // 红色淡 - 神器
  epic: 'hsl(45 50% 97%)',       // 金色淡 - 传说
  rare: 'hsl(210 30% 97%)',      // 银色淡 - 稀有
  common: 'hsl(0 0% 98%)',       // 灰色淡 - 普通
};

// ==================== 基础类型 ====================

export interface CharacterAlias {
  id: string;
  character_id: string;
  alias_type: AliasType;
  content: string;
  order_index: number;
  created_at: string;
  updated_at: string;
}

export interface CardContentItem {
  key: string;
  value: unknown;
  type: 'text' | 'rich_text' | 'list' | 'image' | 'number' | 'boolean';
}

export interface CharacterCard {
  id: string;
  character_id: string;
  title: string;
  icon?: string;
  content: CardContentItem[];
  order_index: number;
  created_at: string;
  updated_at: string;
}

export interface CharacterRelationship {
  id: string;
  character_id: string;
  target_character_id?: string;
  target_name?: string;
  relation_type: RelationType;
  description?: string;
  strength: number;
  is_bidirectional: boolean;
  reverse_description?: string;
  order_index: number;
  target_character?: {
    id: string;
    name: string;
    avatar?: string;
    level: CharacterLevel;
  };
  created_at: string;
  updated_at: string;
}

export interface CharacterArtifact {
  id: string;
  character_id: string;
  name: string;
  quote?: string;
  description?: string;
  artifact_type?: string;
  rarity?: ArtifactRarity;
  image?: string;
  order_index: number;
  created_at: string;
  updated_at: string;
}

// ==================== 人物主类型 ====================

export interface Character {
  id: string;
  project_id: string;
  name: string;
  gender: CharacterGender;
  birth_date?: string;
  birthplace?: string;
  race?: string;
  faction?: string;
  level: CharacterLevel;
  quote?: string;
  avatar?: string;
  full_image?: string;
  first_appearance_volume?: string;
  first_appearance_act?: string;
  first_appearance_chapter?: string;
  last_appearance_volume?: string;
  last_appearance_act?: string;
  last_appearance_chapter?: string;
  order_index: number;
  source?: 'history' | null;
  aliases: CharacterAlias[];
  cards: CharacterCard[];
  relationships: CharacterRelationship[];
  artifacts: CharacterArtifact[];
  created_at: string;
  updated_at: string;
}

// 人物列表项（简化版）
export interface CharacterListItem {
  id: string;
  name: string;
  gender: CharacterGender;
  level: CharacterLevel;
  quote?: string;
  avatar?: string;
  birth_date?: string;
  race?: string;
  faction?: string;
  first_appearance_volume?: string;
  first_appearance_act?: string;
  first_appearance_chapter?: string;
  last_appearance_volume?: string;
  last_appearance_act?: string;
  last_appearance_chapter?: string;
  order_index: number;
  source?: 'history' | null;
  aliases: CharacterAlias[];
  created_at: string;
  updated_at: string;
}

// 人物简要信息（用于选择器）
export interface CharacterSimple {
  id: string;
  name: string;
  avatar?: string;
  level: CharacterLevel;
  source?: 'history' | null;
}

// ==================== 请求/响应类型 ====================

export interface CreateCharacterRequest {
  name: string;
  gender?: CharacterGender;
  birth_date?: string;
  birthplace?: string;
  race?: string;
  faction?: string;
  level?: CharacterLevel;
  quote?: string;
  avatar?: string;
  full_image?: string;
  first_appearance_volume?: string;
  first_appearance_act?: string;
  first_appearance_chapter?: string;
  order_index?: number;
  source?: string;
  aliases?: Omit<CharacterAlias, 'id' | 'character_id' | 'created_at' | 'updated_at'>[];
  cards?: Omit<CharacterCard, 'id' | 'character_id' | 'created_at' | 'updated_at'>[];
  relationships?: Omit<CharacterRelationship, 'id' | 'character_id' | 'target_character' | 'created_at' | 'updated_at'>[];
  artifacts?: Omit<CharacterArtifact, 'id' | 'character_id' | 'created_at' | 'updated_at'>[];
}

export interface UpdateCharacterRequest {
  name?: string;
  gender?: CharacterGender;
  birth_date?: string;
  birthplace?: string;
  race?: string;
  faction?: string;
  level?: CharacterLevel;
  quote?: string;
  avatar?: string;
  full_image?: string;
  first_appearance_volume?: string;
  first_appearance_act?: string;
  first_appearance_chapter?: string;
  last_appearance_volume?: string;
  last_appearance_act?: string;
  last_appearance_chapter?: string;
  order_index?: number;
  source?: string | null;
}

export interface CharacterFilter {
  level?: CharacterLevel;
  search?: string;
  volume?: string;
  act?: string;
  chapter?: string;
  source?: 'history' | 'main' | 'all';
  sort_by?: 'default' | 'order_index' | 'name' | 'created_at' | 'updated_at';
  sort_order?: 'asc' | 'desc';
}

export interface CharacterStats {
  total: number;
  by_level: Record<CharacterLevel, number>;
  by_gender: Record<CharacterGender, number>;
}

export interface BatchDeleteRequest {
  ids: string[];
}

export interface BatchUpdateOrderRequest {
  orders: Record<string, number>;
}

// ==================== 别名相关 ====================

export interface CreateAliasRequest {
  alias_type: AliasType;
  content: string;
  order_index?: number;
}

export interface UpdateAliasRequest {
  alias_type?: AliasType;
  content?: string;
  order_index?: number;
}

// ==================== 卡片相关 ====================

export interface CreateCardRequest {
  title: string;
  icon?: string;
  content?: CardContentItem[];
  order_index?: number;
}

export interface UpdateCardRequest {
  title?: string;
  icon?: string;
  content?: CardContentItem[];
  order_index?: number;
}

// ==================== 关系相关 ====================

export interface CreateRelationshipRequest {
  target_character_id?: string;
  target_name?: string;
  relation_type: RelationType;
  description?: string;
  strength?: number;
  is_bidirectional?: boolean;
  reverse_description?: string;
  order_index?: number;
}

export interface UpdateRelationshipRequest {
  target_character_id?: string;
  target_name?: string;
  relation_type?: RelationType;
  description?: string;
  strength?: number;
  is_bidirectional?: boolean;
  reverse_description?: string;
  order_index?: number;
}

// ==================== 器物相关 ====================

export interface CreateArtifactRequest {
  name: string;
  quote?: string;
  description?: string;
  artifact_type?: string;
  rarity?: ArtifactRarity;
  image?: string;
  order_index?: number;
}

export interface UpdateArtifactRequest {
  name?: string;
  quote?: string;
  description?: string;
  artifact_type?: string;
  rarity?: ArtifactRarity;
  image?: string;
  order_index?: number;
}

// ==================== 快照相关 ====================

export type CharacterSnapshotType = 'volume' | 'act' | 'chapter' | 'custom';

// 快照类型显示名称
export const SnapshotTypeLabels: Record<CharacterSnapshotType, string> = {
  volume: '卷快照',
  act: '幕快照',
  chapter: '章快照',
  custom: '自定义快照',
};

export interface CharacterSnapshot {
  id: string;
  character_id: string;
  snapshot_type: CharacterSnapshotType;
  volume_id?: string;
  act_id?: string;
  chapter_id?: string;
  title: string;
  description?: string;
  attributes: Record<string, unknown>;
  created_at: string;
  updated_at: string;
}

export interface CreateSnapshotRequest {
  snapshot_type: CharacterSnapshotType;
  volume_id?: string;
  act_id?: string;
  chapter_id?: string;
  title: string;
  description?: string;
  attributes?: Record<string, unknown>;
}

export interface UpdateSnapshotRequest {
  snapshot_type?: CharacterSnapshotType;
  volume_id?: string;
  act_id?: string;
  chapter_id?: string;
  title?: string;
  description?: string;
  attributes?: Record<string, unknown>;
}

File: backend/app/schemas/character.py
"""
人物设定系统 - Pydantic Schemas

提供人物角色相关的数据验证和序列化模型
"""

from datetime import datetime
from enum import Enum
from typing import Any, Dict, List, Optional

from pydantic import BaseModel, ConfigDict, Field

# ==================== 枚举定义 ====================


class CharacterLevel(str, Enum):
    """角色等级"""

    PROTAGONIST = "protagonist"  # 主角
    MAJOR_SUPPORT = "major_support"  # 重要配角
    SUPPORT = "support"  # 配角
    MINOR = "minor"  # 小角色


class CharacterGender(str, Enum):
    """角色性别"""

    MALE = "male"  # 男
    FEMALE = "female"  # 女
    OTHER = "other"  # 其他
    UNKNOWN = "unknown"  # 未知


class AliasType(str, Enum):
    """别名类型"""

    ZI = "zi"  # 字
    HAO = "hao"  # 号
    NICKNAME = "nickname"  # 外号
    TITLE = "title"  # 称号
    OTHER = "other"  # 其他


class RelationType(str, Enum):
    """关系类型"""

    FAMILY = "family"  # 亲情
    LOVE = "love"  # 爱情
    FRIEND = "friend"  # 友情
    MASTER = "master"  # 师父
    APPRENTICE = "apprentice"  # 徒弟
    ENEMY = "enemy"  # 敌对
    OTHER = "other"  # 其他


class ArtifactType(str, Enum):
    """器物类型"""

    WEAPON = "weapon"  # 武器
    ARMOR = "armor"  # 防具
    ACCESSORY = "accessory"  # 饰品
    TREASURE = "treasure"  # 法宝
    OTHER = "other"  # 其他


class ArtifactRarity(str, Enum):
    """器物等级"""

    LEGENDARY = "legendary"  # 神器
    EPIC = "epic"  # 传说
    RARE = "rare"  # 稀有
    COMMON = "common"  # 普通


# ==================== 基础 Schema ====================


class CharacterAliasBase(BaseModel):
    """别名基础 Schema"""

    alias_type: AliasType = Field(..., description="别名类型")
    content: str = Field(..., min_length=1, max_length=255, description="别名内容")
    order_index: int = Field(0, ge=0, description="排序索引")


class CharacterAliasCreate(CharacterAliasBase):
    """创建别名 Schema"""

    pass


class CharacterAliasUpdate(BaseModel):
    """更新别名 Schema"""

    alias_type: Optional[AliasType] = None
    content: Optional[str] = Field(None, min_length=1, max_length=255)
    order_index: Optional[int] = Field(None, ge=0)


class CharacterAliasResponse(CharacterAliasBase):
    """别名响应 Schema"""

    model_config = ConfigDict(from_attributes=True)

    id: str
    character_id: str
    created_at: datetime
    updated_at: datetime


# ==================== 卡片内容项 Schema ====================


class CardContentItem(BaseModel):
    """卡片内容项"""

    key: str = Field(..., description="字段名")
    value: Any = Field(..., description="字段值")
    type: str = Field(
        "text", description="字段类型: text|rich_text|list|image|number|boolean"
    )


class CharacterCardBase(BaseModel):
    """卡片基础 Schema"""

    title: str = Field(..., min_length=1, max_length=255, description="卡片标题")
    icon: Optional[str] = Field(None, max_length=100, description="卡片图标")
    content: List[CardContentItem] = Field(default_factory=list, description="卡片内容")
    order_index: int = Field(0, ge=0, description="排序索引")


class CharacterCardCreate(CharacterCardBase):
    """创建卡片 Schema"""

    pass


class CharacterCardUpdate(BaseModel):
    """更新卡片 Schema"""

    title: Optional[str] = Field(None, min_length=1, max_length=255)
    icon: Optional[str] = Field(None, max_length=100)
    content: Optional[List[CardContentItem]] = None
    order_index: Optional[int] = Field(None, ge=0)


class CharacterCardResponse(CharacterCardBase):
    """卡片响应 Schema"""

    model_config = ConfigDict(from_attributes=True)

    id: str
    character_id: str
    created_at: datetime
    updated_at: datetime


# ==================== 人物关系 Schema ====================


class TargetCharacterInfo(BaseModel):
    """目标人物简要信息"""

    id: str
    name: str
    avatar: Optional[str] = None
    level: str = "other"


class CharacterRelationshipBase(BaseModel):
    """关系基础 Schema"""

    target_character_id: Optional[str] = Field(None, description="目标人物ID")
    target_name: Optional[str] = Field(None, max_length=255, description="目标人物名称")
    relation_type: RelationType = Field(..., description="关系类型")
    description: Optional[str] = Field(None, description="关系描述")
    strength: int = Field(50, ge=0, le=100, description="关系强度 0-100")
    is_bidirectional: bool = Field(True, description="是否为双向关系")
    reverse_description: Optional[str] = Field(None, description="反向关系描述")
    order_index: int = Field(0, ge=0, description="排序索引")


class CharacterRelationshipCreate(CharacterRelationshipBase):
    """创建关系 Schema"""

    pass


class CharacterRelationshipUpdate(BaseModel):
    """更新关系 Schema"""

    target_character_id: Optional[str] = None
    target_name: Optional[str] = Field(None, max_length=255)
    relation_type: Optional[RelationType] = None
    description: Optional[str] = None
    strength: Optional[int] = Field(None, ge=0, le=100)
    is_bidirectional: Optional[bool] = None
    reverse_description: Optional[str] = None
    order_index: Optional[int] = Field(None, ge=0)


class CharacterRelationshipResponse(CharacterRelationshipBase):
    """关系响应 Schema"""

    model_config = ConfigDict(from_attributes=True)

    id: str
    character_id: str
    target_character: Optional[TargetCharacterInfo] = None
    created_at: datetime
    updated_at: datetime


# ==================== 器物 Schema ====================


class CharacterArtifactBase(BaseModel):
    """器物基础 Schema"""

    name: str = Field(..., min_length=1, max_length=255, description="器物名称")
    quote: Optional[str] = Field(None, max_length=500, description="器物判词")
    description: Optional[str] = Field(None, description="器物描述")
    artifact_type: Optional[str] = Field(None, max_length=100, description="器物类型（自定义）")
    rarity: Optional[ArtifactRarity] = Field(None, description="器物等级")
    image: Optional[str] = Field(None, max_length=500, description="器物图片URL")
    order_index: int = Field(0, ge=0, description="排序索引")


class CharacterArtifactCreate(CharacterArtifactBase):
    """创建器物 Schema"""

    pass


class CharacterArtifactUpdate(BaseModel):
    """更新器物 Schema"""

    name: Optional[str] = Field(None, min_length=1, max_length=255)
    quote: Optional[str] = Field(None, max_length=500)
    description: Optional[str] = None
    artifact_type: Optional[str] = Field(None, max_length=100)
    rarity: Optional[ArtifactRarity] = None
    image: Optional[str] = Field(None, max_length=500)
    order_index: Optional[int] = Field(None, ge=0)


class CharacterArtifactResponse(CharacterArtifactBase):
    """器物响应 Schema"""

    model_config = ConfigDict(from_attributes=True)

    id: str
    character_id: str
    created_at: datetime
    updated_at: datetime


# ==================== 人物主 Schema ====================


class CharacterBase(BaseModel):
    """人物基础 Schema"""

    name: str = Field(..., min_length=1, max_length=255, description="姓名")
    gender: CharacterGender = Field(CharacterGender.UNKNOWN, description="性别")
    birth_date: Optional[str] = Field(None, max_length=100, description="生辰")
    birthplace: Optional[str] = Field(None, max_length=255, description="出生地")
    race: Optional[str] = Field(None, max_length=100, description="种族")
    faction: Optional[str] = Field(None, max_length=100, description="阵营归属")
    level: CharacterLevel = Field(CharacterLevel.MINOR, description="角色等级")
    quote: Optional[str] = Field(None, description="判词/引言")
    avatar: Optional[str] = Field(None, max_length=500, description="头像URL")
    full_image: Optional[str] = Field(
        None, max_length=500, description="全身形象图片URL"
    )
    first_appearance_volume: Optional[str] = Field(
        None, max_length=100, description="首次出场卷"
    )
    first_appearance_act: Optional[str] = Field(
        None, max_length=100, description="首次出场幕"
    )
    first_appearance_chapter: Optional[str] = Field(
        None, max_length=100, description="首次出场章"
    )
    last_appearance_volume: Optional[str] = Field(
        None, max_length=100, description="最后出场卷"
    )
    last_appearance_act: Optional[str] = Field(
        None, max_length=100, description="最后出场幕"
    )
    last_appearance_chapter: Optional[str] = Field(
        None, max_length=100, description="最后出场章"
    )
    order_index: int = Field(0, ge=0, description="排序索引")


class CharacterCreate(CharacterBase):
    """创建人物 Schema"""

    aliases: List[CharacterAliasCreate] = Field(
        default_factory=list, description="别名列表"
    )
    cards: List[CharacterCardCreate] = Field(
        default_factory=list, description="卡片列表"
    )
    relationships: List[CharacterRelationshipCreate] = Field(
        default_factory=list, description="关系列表"
    )
    artifacts: List[CharacterArtifactCreate] = Field(
        default_factory=list, description="器物列表"
    )


class CharacterUpdate(BaseModel):
    """更新人物 Schema"""

    name: Optional[str] = Field(None, min_length=1, max_length=255)
    gender: Optional[CharacterGender] = None
    birth_date: Optional[str] = Field(None, max_length=100)
    birthplace: Optional[str] = Field(None, max_length=255)
    race: Optional[str] = Field(None, max_length=100)
    faction: Optional[str] = Field(None, max_length=100)
    level: Optional[CharacterLevel] = None
    quote: Optional[str] = None
    avatar: Optional[str] = Field(None, max_length=500)
    full_image: Optional[str] = Field(None, max_length=500)
    first_appearance_volume: Optional[str] = Field(None, max_length=100)
    first_appearance_act: Optional[str] = Field(None, max_length=100)
    first_appearance_chapter: Optional[str] = Field(None, max_length=100)
    last_appearance_volume: Optional[str] = Field(None, max_length=100)
    last_appearance_act: Optional[str] = Field(None, max_length=100)
    last_appearance_chapter: Optional[str] = Field(None, max_length=100)
    order_index: Optional[int] = Field(None, ge=0)


class CharacterListItem(BaseModel):
    """人物列表项 Schema（用于左侧人物栏）"""

    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    gender: str
    level: str
    quote: Optional[str] = None
    avatar: Optional[str] = None
    birth_date: Optional[str] = None
    first_appearance_volume: Optional[str] = None
    first_appearance_act: Optional[str] = None
    first_appearance_chapter: Optional[str] = None
    order_index: int
    # 别名摘要
    aliases: List[CharacterAliasResponse] = []
    created_at: datetime
    updated_at: datetime


class CharacterDetailResponse(CharacterBase):
    """人物详情响应 Schema"""

    model_config = ConfigDict(from_attributes=True)

    id: str
    project_id: str
    aliases: List[CharacterAliasResponse] = []
    cards: List[CharacterCardResponse] = []
    relationships: List[CharacterRelationshipResponse] = []
    artifacts: List[CharacterArtifactResponse] = []
    created_at: datetime
    updated_at: datetime


class CharacterSimpleResponse(BaseModel):
    """人物简要响应 Schema（用于关系选择）"""

    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    level: str
    avatar: Optional[str] = None


# ==================== 筛选和排序 Schema ====================


class CharacterFilter(BaseModel):
    """人物筛选 Schema"""

    level: Optional[CharacterLevel] = Field(None, description="角色等级筛选")
    gender: Optional[CharacterGender] = Field(None, description="性别筛选")
    search: Optional[str] = Field(None, description="搜索关键词（姓名、别名）")
    volume: Optional[str] = Field(None, description="筛选卷")
    act: Optional[str] = Field(None, description="筛选幕")
    chapter: Optional[str] = Field(None, description="筛选章")


class CharacterSort(BaseModel):
    """人物排序 Schema"""

    field: str = Field("order_index", description="排序字段")
    order: str = Field("asc", description="排序方向: asc|desc")


# ==================== 批量操作 Schema ====================


class BatchUpdateOrderRequest(BaseModel):
    """批量更新排序请求"""

    orders: Dict[str, int] = Field(..., description="人物ID到排序索引的映射")


class BatchDeleteRequest(BaseModel):
    """批量删除请求"""

    ids: List[str] = Field(..., description="要删除的人物ID列表")


# ==================== 统计数据 Schema ====================


class CharacterStats(BaseModel):
    """人物统计 Schema"""

    total: int = Field(..., description="人物总数")
    by_level: Dict[str, int] = Field(default_factory=dict, description="按等级统计")
    by_gender: Dict[str, int] = Field(default_factory=dict, description="按性别统计")
